upgrade/upgrade.go

```diff
@@ -82,6 +83,17 @@ func upgrade(currentNodeID, namespace string, config *restclient.Config, lhClien
 	var err error
 	defer cancel()
 
+	// If the current Longhorn is already the latest version,
+	// the leader election & the whole upgrade path could be skipped.
+	lhVersionBeforeUpgrade, err := upgradeutil.GetCurrentLonghornVersion(namespace, lhClient)
+	if err != nil {
+		return err
+	}
+	if semver.Compare(lhVersionBeforeUpgrade, types.CurrentLonghornVersion) >= 0 {
```

Comment:
If the 1st input is invalid or empty, the result will be `-1` :
https://go.dev/play/p/Xr2fwc-t4Nv

Yes, sometimes this would avoid meaningless leader election. But once there are multiple longhorn manager pods starting simultaneously, all of these pods will run this check then trap into the leader election. In order to avoid duplicate upgrades in this case, I add the version comparison in function `doResourceUpgrade` as well.

Reply:
There is a small problem. When both 1st input and 2nd input are invalid, the result is always 0. This mean we won't upgrade when deploying a new dev version
https://go.dev/play/p/0xvHZKt1Xg2

Reply:
Probably. Hence I already add one more check `meta.Version != ""` in the updated commit. PTAL.

Reply:
I was referring to the case when `meta.Version` is `abc` and `lhVersionBeforeUpgrade` is `xyz`. In this case, will we always skip upgrading? If yes, it might be a problem for the developers. 

An other case is when both `meta.Version` and `lhVersionBeforeUpgrade` are `dev-xxx`. In this case, we will skip upgrading. It is a bit inconvenient for the developer.

May be we can do: if the version is dev version, then we always upgrade?  This created another problem too: QA cannot test this PR

Reply:
I just updated the PR. Right now the upgrade will be skipped only when both `meta.Version` and `lhVersionBeforeUpgrade` are valid and match the case.
As for the test, I will help generate multiple images for the testing.

Reply:
Besides, I have refactored the versioning part. Now the source of the version is determined by file `version` in the root directory.

Reply:
Thank you! Great approach